### PR TITLE
New custom auth API

### PR DESCRIPTION
**Pull request by marcogschmidt**
Provide a clear distinction between the two extauth options (Gloo extauth and custom extauth).
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1396

### CODE REVIEW

File: changelog/v0.20.7/update-custom-auth-api.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+- type: NON_USER_FACING
+  description: Provide a clear distinction between the two extauth options (Gloo extauth and custom extauth).
+  issueLink: https://github.com/solo-io/gloo/issues/1396
```
Comment (kdorosh) [inline changelog/v0.20.7/update-custom-auth-api.yaml]: resolvesIssue, false?

Comment (marcogschmidt) [reply changelog/v0.20.7/update-custom-auth-api.yaml]: Yeah, just noticed it. Good catch!

**Comment by marcogschmidt** [comment]
`Panic in Spec Teardown`...

/kick

**Comment by marcogschmidt** [comment]
Access logging e2e test flake

/kick

**Pull request merged**